test/e2e/storage/testsuites/base.go

```diff
@@ -286,10 +287,14 @@ func (r *genericVolumeTestResource) cleanupResource() {
 			}
 			if r.pvc != nil {
 				err := e2epv.DeletePersistentVolumeClaim(f.ClientSet, r.pvc.Name, f.Namespace.Name)
-				framework.ExpectNoError(err, "Failed to delete PVC %v", r.pvc.Name)
+				if err != nil {
+					return errors.Wrapf(err, "Failed to delete PVC %v", r.pvc.Name)
```

Comment:
Is this addressing the issue of cleanups further down not being run?  For example, if we return early here, then storageclass deletion below won't run.